### ISSUE TITLE
Wish list for web interface

### ISSUE THREAD

**Issue by anttinousiainen** [jvde-github/AIS-catcher#77]
The new web interface is super nice addition, thanks for that!

Few minor feature wishes to make it even better:

-  "show/hide all tracks" button. Would be very useful to see by quick glance who is moving and who is not.
- support for own station coordinates to be dynamic, ie. when on a boat. Also auto centering/off-centering based on this would be needed
- 10/20/30min velocity and turning vectors for moving targets
- some way to combine multiple receivers into one web interface

I will think of few more!

**Comment by anttinousiainen** [comment]
And while we are at it, in addition to PPM and RSSI, would be nice to report the received per message AIS-slot number as well. This would be useful in testing the AIS transponders.

**Comment by anttinousiainen** [comment]
> The current way to combine multiple stations is to set up one instance that acts as a UDP server that listens on a particular port (-x .....). The other receivers send their messages with -u to this instance.

Yep, got it working this way, now my rtl is feeding A+B channels and AIrspy C+D.

I assume position for map is taken from A+B channels if available, not from msg 27, yes?

> What needs to be done still is to allow some station identification and pass through of meta data like Rssi and ppm.

Yes this would be great.

**Comment by anttinousiainen** [comment]
> No, all lat/Lon from any message is used. We display the last position coming from any message.

Maybe the lat/lon from msg27 should be lower priority, if there are msg 1/2/3 lat/lon available.

It behaves interestingly otherwise:


![msg27](https://user-images.githubusercontent.com/9679020/218303126-8796fa3d-a40b-480a-b1b5-acdf706e5292.jpg)

**Comment by anttinousiainen** [comment]
> You are right on msg 27. Have not seen the issue you show but it makes logical sense to use this info only when there is not more accurate data.

Yeh, this is somewhat annoying indeed.

![msg01](https://user-images.githubusercontent.com/9679020/218726120-6e2a7a6f-2567-4906-a6b7-e4e771c59772.jpg)
![msg27](https://user-images.githubusercontent.com/9679020/218726130-cf9b260a-7595-4ebe-916f-3ddd84523bfb.jpg)

**Comment by anttinousiainen** [comment]
> How did you envisage btw to get more insights on these AIS slots you mentioned?

Just out of general interest, and also as some older (class B) units seem to have very "interesting" slot allocation mechanism, and seem to be stuck in a same slot for way longer than they are supposed to.

Also would maybe be beneficial to have a -vvv option to show also broken frames and unsuccesfully recovered, partial messages.

**Comment by jortdejong** [comment]
As 'Anttinousiainen' mentioned dynamic 'Own Station Coordinates' should be a great improvement.
(support for own station coordinates to be dynamic, ie. when on a boat. Could be sent from UDP (nmea sentence) or hook directly to gps on serial/usb. Or could be just follow own MMSI, if one has AIS. Also auto centering/off-centering based on this would be needed)

**Comment by anttinousiainen** [comment]
> In the new development branch we have the rudimentary functionality working. Clone it via

Nice progress.

I dunno if its too much, but share_loc on could have an option for mmsi_loc on and then give own ship mmsi to autofollow - or another ship/boat you want to follow from shore?

**Comment by anttinousiainen** [comment]
> Once it looks stable it will be included in the next release.

Great!

btw. Have you given a thought on the receiving of all four channels ABCD at one go? 156.775 MHz to 162.025 MHz requires around 5.5MHz, so it would be doable with many SDRs.

**Comment by anttinousiainen** [comment]
-s 2500k, random minute:
[AIS engine v0.44 #0-0]                 received: 42 msgs, total: 13422 msgs, rate: 0.699505 msg/s

-s 10000k -c ABCD every minute:
[AIS engine v0.44 #0-0]                 received: 0 msgs, total: 0 msgs, rate: 0 msg/s
[AIS engine v0.44 #0-1]                 received: 0 msgs, total: 0 msgs, rate: 0 msg/s

**Comment by anttinousiainen** [comment]
> Is it an idea to run at 10 MHz at AB and CD individually? Might give a hint about interference from nearby frequencies.
> Btw I have been running it for a while now quite successfully on a R2 at 10 MHz.

There is indeed something funny going on with 6MHz and 10MHz. Same decent Diamond antenna I have been using before, and indeed there is almost no reception on higher sample rates. I tried moving the antenna around, no help.
